mask/eolearn/mask/utils.py

```diff
@@ -10,23 +10,72 @@
 This source code is licensed under the MIT license found in the LICENSE
 file in the root directory of this source tree.
 """
+import warnings
+from enum import Enum
+from functools import partial
+from typing import Any, Callable, Optional, Tuple
 
 import cv2
 import numpy as np
-
-
-def map_over_axis(data, func, axis=0):
+from PIL import Image
+
+from eolearn.core.exceptions import EODeprecationWarning
+
+
+class ResizeMethod(Enum):
+    """Methods available for spatial resizing of data."""
+
+    NEAREST = "nearest"
+    LINEAR = "linear"
+    CUBIC = "cubic"
+
+    def get_cv2_method(self, dtype: Any) -> int:
+        """Obtain the constant specifying the interpolation method for the CV2 library."""
+        if np.issubdtype(dtype, np.floating):
```

Comment:
Before this I recommend that you do `dtype` casting like we use [here](https://github.com/sentinel-hub/eo-learn/blob/v1.0.2/core/eolearn/core/utils/common.py#L155). That is because e.g. `np.float32` and `np.dtype('float32')` are not the same things.

Reply:
I recalled that we had some weird shenanigans with dtypes. Didn't feel like it's necessary since the intended use is to pass the dtype of the array, but you're right, its a simple improvement.